client/core/trade.go

```diff
@@ -128,6 +132,26 @@ func (m *matchTracker) matchTime() time.Time {
 	return time.UnixMilli(int64(m.MetaData.Proof.Auth.MatchStamp)).UTC()
 }
 
+func (m *matchTracker) swapSpentAgo() time.Duration {
+	now := time.Now() // avoid system clock with mutex locked
```

Comment:
> `// avoid system clock with mutex locked`

Why?

Reply:
It's a relatively expensive syscall, and there have been Go bugs that have made it worse.  This isn't a hot path, so perhaps an unnecessary optimization.

https://github.com/golang/go/issues/22601
https://tpaschalis.github.io/golang-time-now/